src/Utils/request/query.ts

```diff
@@ -5,15 +5,26 @@ import { getResponseBody } from "@/Utils/request/request";
 import { QueryOptions, Route } from "@/Utils/request/types";
 import { makeHeaders, makeUrl } from "@/Utils/request/utils";
 
+// Extend the QueryOptions interface to include customHeaders
+export interface ExtendedQueryOptions<TBody> extends QueryOptions<TBody> {
+  customHeaders?: Record<string, string>;
+}
+
 async function queryRequest<TData, TBody>(
   { path, method, noAuth }: Route<TData, TBody>,
-  options?: QueryOptions<TBody>,
+  options?: ExtendedQueryOptions<TBody>,
 ): Promise<TData> {
   const url = `${careConfig.apiUrl}${makeUrl(path, options?.queryParams, options?.pathParams)}`;
 
+  // Merge customHeaders with default headers
+  const headers = {
+    ...makeHeaders(noAuth ?? false),
+    ...(options?.customHeaders || {}),
+  };
```

Comment:
We could modify the `makeHeaders` function directly to concat the headers through Headers obj. initialization itself.

```suggestion
  const headers = makeHeaders(noAuth ?? false, additionalHeaders)
```

Reply:
Made the changes as suggested